scripts/create_framework.sh

```diff
@@ -0,0 +1,96 @@
+# if there is no command line ask for a version number
+if [ -z $1  ]; then
+    # 2 get the verison number
+    # get the verison number
+    # at this point the version in the project file should be correct, so use it.
+    project_file="./SmartDeviceLink-iOS.xcodeproj/project.pbxproj"
+    current_version_number=$(sed -n '/MARKETING_VERSION/{s/MARKETING_VERSION = //;s/;//;s/^[[:space:]]*//;p;q;}' $project_file)
+    if [ -z $current_version_number ]; then current_version_number="1.0.0"; fi
+    echo "Current Version: "$current_version_number
+
+    # todo - we can streamline this by trusting the project file to always have the correct version (bail out if project file missing)
+    prompt_user "Is this version correct"
+    if [[ $? == 0 ]]; then
+        # Prompt user for new version
+        echo "Enter the new version number (semantic versioning x.x.x format) or blank to skip: "
+        read new_version_number
+
+        # If blank or the same, then skip. Otherwise change the version number
+        if [ -z $new_version_number ]; then
+            echo "No version number entered. Skipping..."
+            new_version_number=$current_version_number
+        fi
+    fi
+else
+    new_version_number=$1
+fi
+# 3 Add a binary xcframework archive for manual installation with the following commands
+echo
+echo "Creating a binary xcframework for manual installation"
+# this builds the framework
+xcodebuild archive -project 'SmartDeviceLink-iOS.xcodeproj/' -scheme 'SmartDeviceLink' -configuration Release -destination 'generic/platform=iOS' -archivePath './SmartDeviceLink-Device.xcarchive' SKIP_INSTALL=NO
+xcodebuild archive -project 'SmartDeviceLink-iOS.xcodeproj/' -scheme 'SmartDeviceLink' -configuration Release -destination 'generic/platform=iOS Simulator' -archivePath './SmartDeviceLink-Simulator.xcarchive' SKIP_INSTALL=NO
+xcodebuild -create-xcframework -framework './SmartDeviceLink-Device.xcarchive/Products/Library/Frameworks/SmartDeviceLink.framework/' -framework './SmartDeviceLink-Simulator.xcarchive/Products/Library/Frameworks/SmartDeviceLink.framework/' -output './SmartDeviceLink.xcframework'
+
+# todo - is there a way we can test that the build was successful.
+
+folder="SmartDeviceLink.xcframework"
+zip_file_name="SmartDeviceLink-$new_version_number.xcframework.zip"
+# kill the old zip if present.  Useful for re-running the script
+if [ -f $zip_file_name ]; then rm $zip_file_name; fi
+# verify folder exists before acting on it.
+if [ -d "$folder" ]; then 
+    zip $zip_file_name $folder
+    # Check to see if the zip exists, and then remove old files.
+    if [ -f "$zip_file_name" ]; then rm -r $folder; fi
+fi
+
+#cleanup artifacts
+folder="SmartDeviceLink-Device.xcarchive"
+if [ -d "$folder" ]; then rm -r $folder; fi
+
+folder="SmartDeviceLink-Simulator.xcarchive"
+if [ -d "$folder" ]; then rm -r $folder; fi
+
+echo
+echo "The xcframework zip file was created at $zip_file_name. Please add it to the Github Release, then press enter..."
```

Comment:
```suggestion
echo "The xcframework zip file was created at $zip_file_name."
```

Because this isn't a "release" script, this shouldn't say this here. It should be moved to the release script when the creation finishes.

Reply:
The deleted section was not moved to the release script.